api/allocations_exec.go

```diff
@@ -177,15 +183,19 @@ func (s *execSession) startTransmit(ctx context.Context, conn *websocket.Conn) <
 
 	// send a heartbeat every 10 seconds
 	go func() {
+		t := time.NewTimer(heartbeatInterval)
```

Comment:
`helpers/` not available to `api/`, so no helper function here

- make sure we call `t.Stop`
- make sure the interval can never be non-positive